### PR TITLE
feature: remove TextInput border-bottom with hideLine prop

### PR DESCRIPTION
**Pull request by kialysa**
Currently, TextInput and PhoneInput component automatically has a border-bottom due to ListItem.
The changes will allow the passing of hideLine prop to trigger ListItem hideLine.

Checklist:

- [ ] Test added / Snapshots updated
- [ ] Documentation added / updated

### CODE REVIEW

File: src/molecules/TextInput.jsx [hunk 1]
```diff
@@ -160,7 +160,11 @@ class TextInput extends React.Component {
     return (
       <Theme>
         {({ theme, colorize }) => (
-          <ListItem padded={false} backgroundColor={colorize(backgroundColor)}>
+          <ListItem
+            padded={false}
+            backgroundColor={colorize(backgroundColor)}
+            {...props}
```
Comment (dan-lee) [inline src/molecules/TextInput.jsx]: Be careful with spreading all the props, the same props are also spread on the input field. Now the div has a `placeholder`. (see snapshot)

File: src/behaviour/__snapshots__/FormValidityProvider.test.jsx.snap [hunk 2]
```diff
@@ -180,6 +180,9 @@ exports[`Check FormValidityProvider should click on the button and get a custom
   <div
     data-css-dau5wj=""
     data-css-h7582n=""
+    name="email"
+    placeholder="Your email"
```
Comment (dan-lee) [inline src/behaviour/__snapshots__/FormValidityProvider.test.jsx.snap]: ^

File: src/molecules/TextInput.jsx (hunk 1 above)
Comment (kialysa) [reply src/molecules/TextInput.jsx]: yikes. thanks for pointing that out!

**Comment by mosch** [comment]
What's your actual need for hiding the line? I think putting the TextInput into a ListItem was not the best decision in the first place. So maybe it's the right moment to split things apart?

If not, I would go for "withLine" or "showLine" (default true) to get to consistent "always positive" prop-types.

**Comment by mosch** [comment]
See: #156

**Comment by mosch** [comment]
closing this in fav of https://github.com/allthings/elements/pull/156

**Pull request closed without merging**